### ISSUE TITLE
Add a flag as to whether to display the custom extension in extension manager

### ISSUE THREAD

**Issue by ggbhat** [jupyterlab/jupyterlab#7256]
As developer wants to add flag in extension package metadata so that extension will not be visible in extension manager and that can not uninstall/disable by user.

Suppose some customization is made to jupyter by adding custom extension for example adding custom theme,session handling then we should avoid user from uninstalling  extension from manager but the same time he should be able to install contributed extensions.

Make use of core flag or add new flag to control display of extension in extension manager

**Comment by vidartf** [comment]
There are two cases here:
1. The general case of extensions that are installed by the user.
2. The case where someone is creating a deployment of JupyterLab, and want to add an extension to the set of "core" extensions that uses cannot modify.

For 1., I think extensions should always be visible. As such, I don' think this flag/info should exist in the package metadata, but rather it should be in the lab configuration. This should be possible to implement, but isn't currently exposed via any CLI commands/options. The work on `CoreConfig`/`AppOptions` should at least have made this easier to achieve in the future.

One alternatives for now:
- Modify the package.json file in `jupyterlab/staging` in your python deployment, to include your extensions. Note that this will be overwritten on any update of jupyterlab, so it is a limited workaround.

**Comment by ggbhat** [comment]
@vidartf  yes in case of 2, suppose jupyter lab is distributed as docker instance and some custom extensions are added but want to make it set of core extensions. 
Can you explain how we can update jupyterlab/staging in case of docker environment.

**Comment by ggbhat** [comment]
@vidartf  @jasongrout  can anyone help me on this..

**Comment by vidartf** [comment]
The most direct, but least "stable" (as it is not a part of an API) is to simply patch this file in your deployment: https://github.com/jupyterlab/jupyterlab/blob/master/jupyterlab/staging/package.json

A larger effort would be to build another Python app on top of the lab apps, that modifies the `CoreConfig` object as phoila does: https://github.com/vidartf/phoila/blob/master/phoila/commands.py

We probably want a story in the future to make this easier to do, but we are unfortunately not there yet. One way would probably be to ensure that the `CoreConfig` object on the various commands is configurable, and write a tutorial on how to configure that (via a python config file probably).
